Redesign of music sneak peak and fix MarqueeTextView issue

Al3Gr opened the pull request:
Redesign of music sneak peak.

I also fix MarqueeTextView issue that presents when animation is running and the user changes song


https://github.com/user-attachments/assets/de9dc110-e06e-45d8-80fb-42e4eaffe674

Alexander5015 commented:
Thanks for your contribution, @Al3Gr! I’m currently unable to accept PRs due to a licensing issue, but I’ll review it once it’s resolved. Also, please note that we’ll be changing our license to GPL3.

Al3Gr commented:
@Alexander5015 thanks for your feedback! No problem regarding the change of license.

In the last commit I add a new option in Media Settings for enabling the new UI of sneak peek (the two toggles are mutually exclusive, as shown in the image bellow). In this way if there are some functions linked to the old sneak peak I don't mess with them.
<img width="576" alt="Screenshot 2025-04-11 alle 17 06 16" src="https://github.com/user-attachments/assets/7b813b0f-a5f0-4b72-a164-309fc5652103" />

Al3Gr commented:
@Alexander5015 I have applied the changes you requested.

<img width="574" alt="Screenshot 2025-04-15 alle 21 57 14" src="https://github.com/user-attachments/assets/3f8b1d3c-a2de-406d-933c-5ec738704971" />
<img width="569" alt="Screenshot 2025-04-15 alle 21 57 21" src="https://github.com/user-attachments/assets/c23230d0-6bec-45ac-89ef-4654b4856f30" />

Alexander5015 commented:
Awesome, this looks great! It seems like there is a merge conflict right now. I still haven't fixed the licensing but fixing MarqueeText is important so I am going to review and merge under current license if the aren't any problems and change the license later.

Al3Gr commented:
@Alexander5015 the merge conflict is in the conditionalModifier (as shown in the image below) because I added an extra condition to prevent a blink effect when the user selects the inline style. You can try it yourself by simply removing the condition I added.

<img width="885" alt="Screenshot 2025-04-15 alle 22 11 13" src="https://github.com/user-attachments/assets/2ea42bfc-fe9f-4d22-925d-a22e8a02c7ce" />

Alexander5015 commented:
@Al3Gr Can you resolve the merge conflict? Also, if you are on our discord send me a message and I will add you to the contributors.

Al3Gr commented:
@Alexander5015 conflict resolved. I am in the discord server (nickname is Al3Gr). Where I send you a message? In some channels of boring.notch server or in private?

Alexander5015 commented:
It's ok, I added you @Al3Gr

Alexander5015 commented:
I fixed some changes that accidentally got reverted by the merge and checked out your PR. It also seems like the inline sneak peek doesn't work on play/pause.

Al3Gr commented:
@Alexander5015 I fixed sneak peek on play/pause case.

Alexander5015 commented:
Okay, I’ll take a deeper look later today. I’ll probably rearrange things so that both sneak peeks use the same entry function to make maintenance easier later. Then, I’ll merge this. Great work!
